src/main/scala/io/iohk/ethereum/vm/DataWord.scala

```diff
@@ -0,0 +1,90 @@
+package io.iohk.ethereum.vm
+
+import akka.util.ByteString
+
+
+object DataWord {
+
+  val MaxLength = 32
+
+  val MaxWord: BigInt = BigInt(2).pow(MaxLength * 8) - 1
+
+  val Zeros: ByteString = ByteString(Array.fill[Byte](MaxLength)(0))
+
+  def apply(value: ByteString): DataWord = {
+    require(value.length <= MaxLength, s"Input byte array cannot be longer than $MaxLength: ${value.length}")
+    DataWord(value.foldLeft(BigInt(0)){(n, b) => (n << 8) + (b & 0xff)})
+  }
+
+  def apply(n: BigInt): DataWord = {
+    new DataWord(fixBigInt(n))
+  }
+
+  def apply[N: Integral](n: N): DataWord = {
+    val num = implicitly[Integral[N]]
+    apply(BigInt(num.toLong(n)))
+  }
+
+  def fixBigInt(n: BigInt): BigInt = {
+    if (n == -MaxWord) {
+      0
+    } else if (n < 0) {
+      n % MaxWord + MaxWord
+    } else {
+      n % MaxWord
+    }
+  }
+
+}
+
+/** Stores 256 bit words and adds a few convenience methods on them.
+ *  Internally a word is stored as a BigInt. */
+class DataWord private (private val n: BigInt) {
```

Comment:
can this be case class and that way we can avoid adding our own `equals` and `hashCode` ?